scipy/optimize/optimize.py

```diff
@@ -3061,6 +2904,580 @@ def show_options(solver=None, method=None, disp=True):
         return text
 
 
+class Optimizer(object):
+    """
+    Class based minimization of scalar functions of one or more variables::
+
+        minimize f(x) subject to
+
+        g_i(x) >= 0,  i = 1,...,m
+        h_j(x)  = 0,  j = 1,...,p
+
+    where x is a vector of one or more variables.
+    ``g_i(x)`` are the inequality constraints.
+    ``h_j(x)`` are the equality constraints.
+
+    Optionally, the lower and upper bounds for each element in x can also be
+    specified using the `bounds` argument.
+    """
+    opt_options = {'args': (), 'bounds': None, 'x0': None, 'jac': None,
+                   'hess': None, 'maxiter': np.inf, 'maxfun': np.inf, 'disp': False,
+                   'callback': None}
+
+    def __init__(self, func, **options):
+        self.opt_options.update(options)
+        if isinstance(func, OptimizeFunction):
```

Comment:
Do we want to issue a deprecation warning here if `args`, `jac` or `hess` specified?

Either way we should mention the API change in the docstring.

Reply:
Removed jac, args and hess. This is a new interface, better to be clean to start with. People will still be able to use the old `minimize` interface for a long time yet.